frontend/app/src/premium/premium-apis.ts

```diff
@@ -46,17 +47,20 @@ export const assetsApi = (): AssetsApi => {
   const { getNftDetails } = useNftAssetInfoStore();
 
   return {
-    assetInfo: identifier => {
-      const nft = get(getNftDetails(identifier)) as AssetInfo | null;
-      return nft ?? get(assetInfo(identifier));
-    },
-    assetSymbol: identifier => {
-      const nft = get(getNftDetails(identifier)) as AssetInfo | null;
-      return nft?.symbol ?? get(assetSymbol(identifier));
-    },
-    tokenAddress: identifier => {
-      return get(tokenAddress(identifier));
-    }
+    assetInfo: (identifier: MaybeRef<string>) =>
+      computed(() => {
+        const nft = get(getNftDetails(identifier)) as AssetInfo | null;
+        return nft ?? get(assetInfo(identifier));
+      }),
+    assetSymbol: (identifier: MaybeRef<string>) =>
+      computed(() => {
+        const nft = get(getNftDetails(identifier)) as AssetInfo | null;
+        return nft?.symbol ?? get(assetSymbol(identifier));
+      }),
+    tokenAddress: (identifier: MaybeRef<string>) =>
+      computed(() => {
+        return get(tokenAddress(identifier));
```

Comment:
at least in this case tokenAddress returns a computed right? so we could just `=> tokenAddress(identifier)`